tests/mocks/MockEmailService.ts

```diff
@@ -0,0 +1,30 @@
+import { mock } from "bun:test";
+
+// A mock for the cookies function in the NextJS next/header module.
+export class MockEmailService {
+  private haveSendError: boolean;
+
+  constructor() {
+    this.haveSendError = false;
+  }
+
+  setSendError(error: boolean) {
+    this.haveSendError = error;
+  }
+
+  // Applies this mock to be the cookies used by the next/header module. This method
+  // must be called in order for this mock to be applied.
+  async apply(): Promise<void> {
+    await mock.module("@good-dog/email/email-service", () => ({
+      default: {
+        setApiKey: () => {
```

Comment:
I'll go over this at standup, but I think we can improve the interface for this mock

We would like to be able to do something like

```ts
const mockEmailService = new MockEmailService()

beforeAll(async () => {
    await mockEmailService.apply()
})

afterEach(() => {
   mockEmailService.clear()
})

test("my test", () => {
   // do something
   expect(mockEmailService.send).toBeCalled()
   expect(mockEmailService.send).toBe(true) // or some other property
})


```
  That way in our code we can do something like